### PR TITLE
Improvement: Custom scrollbar

### PR DESCRIPTION
**Pull request by alexis-ld**
**Component**: **Layout**

**Description**:

This enables scrollbar styling for every children within the **Layout** component.
We wanted to have a thinner scrollbar fitting the design better.

**For Chrome/Safari/Edge** we use the non-standard ::webkit-scrollbar CSS properties which are well supported across these browsers. The scrollbar is 4px wide and has a very small left padding to avoid the scrollbar being directly sticked to the scrollable data.

Before : 
chrome_before
After:
chrome_after

**For Firefox** we use the scrollbar-width and scrollbar-color properties. However these give less option than webkit and the only options for scrollbar-width are 'auto | thin | none' . There is no way to specify a pixel width for now. This has been set to 'thin', it is slightly wider than its webkit counterpart but is still way smaller than the original one.

Before :
firefox_before
After :
firefox_after

**Breaking Changes**: none

---

### CODE REVIEW

File: src/lib/components/layout/Layout.component.js [hunk 1]
```diff
@@ -1,24 +1,64 @@
...
+          display: none;
+        }
+        ::-webkit-scrollbar-corner {
+          background-color: transparent;
+        }
+
+        // Firefox
+        scrollbar-color: ${brand.border} ${brand.primary};
+        scrollbar-width: thin;
+      }
```
Comment (JBWatenbergScality) [inline src/lib/components/layout/Layout.component.js]: Ok for the approach ! But I think we should add it to `/src/lib/index.css` in order to make it available to applications that don't use this `Layout` component (we may have this use case for embedded applications)

Comment (alexis-ld) [reply src/lib/components/layout/Layout.component.js]: @JBWatenbergScality Ok, make sense. Do you have any suggestions on how to use the color variables (in this case primary and border) in a .css file ?

**Comment by Cuervino** [comment]
Nice. On the screenshot, I like the Firefox one. 
It looks like the Chrome version is cut, especially as the round effect at the end is not complete. Maybe it's just a feeling.

Something like that:
image

**Comment by alexis-ld** [comment]
> Nice. On the screenshot, I like the Firefox one.
> It looks like the Chrome version is cut, especially as the round effect at the end is not complete. Maybe it's just a feeling.
> 
> Something like that:
> image

@Cuervino indeed, probably linked to the background-clip :) will fix it

File: src/lib/components/layout/Layout.component.js (hunk 1 above)
Comment (JBWatenbergScality) [reply src/lib/components/layout/Layout.component.js]: I see 2 options : 
1. Create a Scrollbar component independent from the layout and document that it should be used as a top level wrapper if we want an UI to be able to benefit from this scrollbar. For this approach restraining the `*` matcher to the root of this wrapper component could make sense
2. Introduce a theme component wrapper that, when introduced in the app, define css variables (https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) for each of the theme color. We then would be able to use those css variables directly in `index.css` 

IMO the first one will be easier to maintain and brings some benefits (decoupling the custom navbar to anything else) despite forcing us to add a wrapper in the UI where we want this custom navbar (but I think its ok and makes it explicit if one day we wonder where does this customisation comes from).

**Comment by alexis-ld** [comment]
@ChengYanJin @JBWatenbergScality Rebased and moved the Scrollbar rules to their own wrapper component.

**Review by JBWatenbergScality — APPROVED** [review]

**Review by ChengYanJin — COMMENTED** [review]
Hello @alexis-ld 
The changes look good to me!
Maybe add a note in the README to explain what change should we proceed with in order to have the styled scrollbar in all the components?

**Review by ChengYanJin — APPROVED** [review]
LGTM! Thanks for adding the README to explain how to use the customized scrollbar.

**Pull request merged**